enterprise/server/usage/usage.go

```diff
@@ -79,16 +78,24 @@ var (
 )
 
 type TrackerOpts struct {
+	// TEST ONLY options
+
+	// Clock overrides the clock used by the usage tracker.
 	Clock timeutil.Clock
+
+	// DisableRedlock disables the Redis locking mechanism used to prevent
+	// flush operations from accessing the DB concurrently.
+	DisableRedlock bool
```

Comment:
Test only options that disable a big chunk of the impl are kinda gross -- could we pass in a distributed lock interface in the constructor and then implement that with a regular old lock or no-ops for testing?

type DistributedLock interface {
  Lock(ctx context.Context, key string, expiry time.Duration)
  Unlock(ctx context.Context, key string)
}

Reply:
Done